yt_dlp/extractor/dailymotion.py

```diff
@@ -232,16 +297,35 @@ def _extract_embed_urls(cls, url, webpage):
         for mobj in re.finditer(
                 r'(?s)DM\.player\([^,]+,\s*{.*?video[\'"]?\s*:\s*["\']?(?P<id>[0-9a-zA-Z]+).+?}\s*\);', webpage):
             yield from 'https://www.dailymotion.com/embed/video/' + mobj.group('id')
```

Comment:
It's unlikely that this was meant to yield `'h'`, `'t'`, `'t'`, etc:  
```suggestion
            yield 'https://www.dailymotion.com/embed/video/' + mobj.group('id')
```

Reply:
I agree. Let's fix it in the upcoming cleanup PR.

Reply:
Also, the `metadata` key that I see with thumbnails is `thumbnails` rather than `posters`, and then the thumbnail URLs don't quite match (the penultimate path components differ).

Reply:
the traversal path branches to collect from both `posters` and `thumbnails`

Reply:
as for the `_extract_embed_urls` bug, this wasn't touched in this PR, none of our tests cover this code path, and it's dead code for all we know